src/Devices/GateReader/GateReaderDevice.cs

```diff
@@ -67,11 +88,16 @@ public GateDirection Direction
         /// <returns></returns>
         public async Task SetDirectionAsync(GateDirection value)
         {
-            // set device twin property
-            await this._DeviceClient.SetDigitalTwinPropertyAsync(new KeyValuePair<string, object>("GateDirection", value.ToString()));
 
-            // set local cached value. Done after the device twin update so if it failed, we didn't update our local value
-            this.CurrentDirection = value;
+            // if the direction is different then what we've had before...
+            if (this.Direction != value)
+            {
+                // set device twin property
+                await this._DeviceClient.SetDigitalTwinPropertyAsync(new KeyValuePair<string, object>("GateDirection", value.ToString()));
```

Comment:
if I recall, its needed because the 'object' needed to be a string so that it can go over the wire properly. One could argue that it should be a <string, string> generic, but I believe it requires a <string, object>